src/main/java/org/springframework/data/couchbase/core/ReactiveCouchbaseTemplate.java

```diff
@@ -0,0 +1,411 @@
+ * limitations under the License.
+ */
+package org.springframework.data.couchbase.core;
+
+import static org.springframework.data.couchbase.core.CouchbaseTemplate.ensureNotIterable;
+
+
+import com.couchbase.client.java.AsyncBucket;
+import com.couchbase.client.java.Bucket;
+import com.couchbase.client.java.PersistTo;
+import com.couchbase.client.java.ReplicateTo;
+import com.couchbase.client.java.cluster.ClusterInfo;
+import com.couchbase.client.java.document.Document;
+import com.couchbase.client.java.document.RawJsonDocument;
+import com.couchbase.client.java.document.json.JsonObject;
+import com.couchbase.client.java.error.DocumentDoesNotExistException;
+import com.couchbase.client.java.query.*;
+import com.couchbase.client.java.view.*;
+import org.springframework.data.couchbase.core.convert.CouchbaseConverter;
+import org.springframework.data.couchbase.core.convert.MappingCouchbaseConverter;
+import org.springframework.data.couchbase.core.convert.translation.JacksonTranslationService;
+import org.springframework.data.couchbase.core.convert.translation.TranslationService;
+import org.springframework.data.couchbase.core.mapping.*;
+import org.springframework.data.couchbase.core.query.Consistency;
+import org.springframework.data.couchbase.core.query.reactor.mapping.*;
+import org.springframework.data.couchbase.core.support.TemplateUtils;
+import org.springframework.data.mapping.PersistentPropertyAccessor;
+import org.springframework.data.mapping.context.MappingContext;
+import org.springframework.data.mapping.model.ConvertingPropertyAccessor;
+import org.springframework.data.repository.util.ReactiveWrapperConverters;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import rx.Observable;
+
+/**
+ * ReactiveCouchbaseTemplate implements operations supporting reactor types
+ *
+ * @author Subhashni Balakrishnan
+ * @since 3.0
+ */
+public class ReactiveCouchbaseTemplate implements ReactiveCouchbaseOperations {
+
+    private Bucket syncClient;
+    private AsyncBucket client;
+    private final ClusterInfo clusterInfo;
+    private final CouchbaseConverter converter;
+    private final TranslationService translationService;
+    protected final MappingContext<? extends CouchbasePersistentEntity<?>, CouchbasePersistentProperty> mappingContext;
+    private Consistency configuredConsistency = Consistency.DEFAULT_CONSISTENCY;
+    private static final WriteResultChecking DEFAULT_WRITE_RESULT_CHECKING = WriteResultChecking.NONE;
+
+    private WriteResultChecking writeResultChecking = DEFAULT_WRITE_RESULT_CHECKING;
+
+
+    @SuppressWarnings("unchecked")
+    public <T> Mono<T> save(T objectToSave) {
+        return ReactiveWrapperConverters.toWrapper(doPersist(objectToSave, PersistTo.NONE, ReplicateTo.NONE), Mono.class);
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T> Flux<T> save(Iterable<T> batchToSave) {
+        return Flux.fromIterable(batchToSave)
```

Comment:
Batch saving pretends some sort of atomic operation but in in fact each item is saved individually. I propose to shift that responsibility to the user code so users can decide upon batch sizes, call `save(T)` on their own, etc.